src/pudl/extract/csv.py

```diff
@@ -1,72 +1,121 @@
 """Extractor for CSV data."""
-from csv import DictReader
-from importlib import resources
-from zipfile import ZipFile
-
 import pandas as pd
 
 import pudl.logging_helpers
+from pudl.extract.extractor import GenericExtractor
 
 logger = pudl.logging_helpers.get_logger(__name__)
 
 
-def open_csv_resource(dataset: str, base_filename: str) -> DictReader:
-    """Open the given resource file as :class:`csv.DictReader`.
+class CsvExtractor(GenericExtractor):
+    """Class for extracting dataframes from CSV files.
 
-    Args:
-        dataset: used to load metadata from package_data/{dataset} subdirectory.
-        base_filename: the name of the file in the subdirectory to open.
+    The extraction logic is invoked by calling extract() method of this class.
     """
-    csv_path = resources.files(f"pudl.package_data.{dataset}") / base_filename
-    return DictReader(csv_path.open())
-
 
-def get_table_file_map(dataset: str) -> dict[str, str]:
-    """Return a dictionary of table names and filenames for the dataset.
-
-    Args:
-        dataset: used to load metadata from package_data/{dataset} subdirectory.
-    """
-    return {
-        row["table"]: row["filename"]
-        for row in open_csv_resource(dataset, "table_file_map.csv")
-    }
+    def __init__(self, ds, dataset_name):
+        """Create a new instance of CsvExtractor.
 
+        This can be used for retrieving data from CSV files.
 
-class CsvExtractor:
-    """Generalized class for extracting dataframes from CSV files.
+        Args:
+            ds (datastore.Datastore): An initialized datastore, or subclass
+            dataset_name: Name of the package/dataset to load
+        """
+        self.ds = ds
+        self._dataset_name = dataset_name
 
-    The extraction logic is invoked by calling extract() method of this class.
-    """
+    def extract(self, **partition):
+        """Extracts dataframes.
 
-    def __init__(self, zipfile: ZipFile, table_file_map: dict[str, str]):
-        """Create a new instance of CsvExtractor.
+        Returns dict where keys are page names and values are
+        DataFrames containing data across given years.
```

Comment:
I need to double check this. We'll need to add a `year` column at some point to distinguish the source data, which is time-variant, and it isn't clear to me if/where that is happening.

Reply:
Right now the year is getting passed into the extract function as part of `**partition`. We've switched to unpacking this to accommodate a few datasets that require more than one partition to point to a particular file (e.g., `year` _and_ `form`). Most but not all datasets use `year` as the partition, so we should avoid baking `year` into the extraction where possible, although it's certainly baked in currently (see, e.g. https://github.com/catalyst-cooperative/pudl/issues/3366).